athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -110,22 +109,35 @@ protected ElasticsearchRecordHandler(AmazonS3 amazonS3, AWSSecretsManager secret
     protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recordsRequest,
                                       QueryStatusChecker queryStatusChecker)
     {
-        logger.info("readWithConstraint - enter\n\nDomain: {}\n\nIndex: {}\n\nMapping: {}\n\nConstraints: {}",
+        logger.info("readWithConstraint - enter\n\nDomain: {}\n\nIndex: {}\n\nMapping: {}",
                 recordsRequest.getTableName().getSchemaName(), recordsRequest.getTableName().getTableName(),
-                recordsRequest.getSchema(), recordsRequest.getConstraints().getSummary());
+                recordsRequest.getSchema());
 
-        String endpoint = ElasticsearchHelper.getDomainEndpoint(recordsRequest.getTableName().getSchemaName());
+        String domain = recordsRequest.getTableName().getSchemaName();
+        String endpoint = recordsRequest.getSplit().getProperty(domain);
         long numRows = 0;
-        AtomicLong numResultRows = new AtomicLong(0);
 
-        if (!endpoint.isEmpty() && queryStatusChecker.isQueryRunning()) {
+        if (endpoint != null && queryStatusChecker.isQueryRunning()) {
             AwsRestHighLevelClient client = clientFactory.getClient(endpoint);
             try {
+                // Create field extractors for all data types in the schema.
+                GeneratedRowWriter.RowWriterBuilder builder =
+                        GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+                for (Field field : recordsRequest.getSchema().getFields()) {
+                    Extractor extractor = typeUtils.makeExtractor(field);
+                    if (extractor != null) {
+                        builder.withExtractor(field.getName(), extractor);
+                    }
+                    else {
+                        builder.withFieldWriterFactory(field.getName(), typeUtils.makeFactory(field));
+                    }
+                }
+                GeneratedRowWriter rowWriter = builder.build();
```

Comment:
can we move this to its own function so we make this function smaller?

Reply:
Moved to own function - createFieldExtractors(...)